src/Portage/EBuild.hs

```diff
@@ -118,17 +102,15 @@ showEBuild now ebuild =
   ss ("# ebuild generated by hackport " ++ hackportVersion ebuild). nl.
   sconcat (map (\(k, v) -> ss "#hackport: " . ss k . ss ": " . ss v . nl) $ used_options ebuild).
   nl.
+  (case cabal_pn ebuild of
+     Nothing -> id
+     Just pn -> ss "CABAL_PN=". quote pn. nl.).
+  nl.
   ss "CABAL_FEATURES=". quote' (sepBy " " $ map render (features ebuild)). nl.
   ss "inherit haskell-cabal". nl.
   nl.
-  (case my_pn ebuild of
-     Nothing -> id
-     Just pn -> ss "MY_PN=". quote pn. nl.
-                ss "MY_P=". quote "${MY_PN}-${PV}". nl.
-                ss "S=". quote ("${WORKDIR}/${MY_P}"). nl. nl).
```

Comment:
As per build failure: can we remove these outermost parentheses? I have a feeling they may be redundant.